Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/TimeslotController.java

```diff
@@ -40,22 +43,27 @@ public ResponseEntity<TimeslotDto> getById(@PathVariable long id){
         return new ResponseEntity<>(timeslotDtoMapper.fromTimeslotToTimeslotDto(timeslot), HttpStatus.OK);
     }
 
-    @PostMapping(value = "/")
-    public ResponseEntity<TimeslotDto> save(@RequestBody TimeslotDto timeslotNew){
-        Timeslot timeslot = timeslotService.save(timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotNew), timeslotNew.personId, timeslotNew.courtId);
-
-        return new ResponseEntity<>(timeslotDtoMapper.fromTimeslotToTimeslotDto(timeslot), HttpStatus.CREATED);
+    @PostMapping("/")
+    public ResponseEntity<TimeslotResponseDto> save(@RequestBody TimeslotDto timeslotNew){
+        TimeslotResponseDto timeslotResponse = timeslotResponseDtoMapper.toTimeslotResponseDto(timeslotService.reserveTimeslot(timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotNew)));
+        if (timeslotResponse.timeslot != null)
+            return new ResponseEntity<>(timeslotResponse, HttpStatus.CREATED);
+        else
+            return new ResponseEntity<>(timeslotResponse, HttpStatus.BAD_REQUEST);
     }
 
     @PutMapping(value = "/{id}")
-    public ResponseEntity<TimeslotDto> update(@PathVariable long id, @RequestBody TimeslotDto timeslotUpdate){
+    public ResponseEntity<TimeslotResponseDto> update(@PathVariable long id, @RequestBody TimeslotDto timeslotUpdate){
         Timeslot timeslotExisting = timeslotService.getById(id);
         if (timeslotExisting == null) {
             return new ResponseEntity<>( HttpStatus.BAD_REQUEST);
```

Comment:
Maybe to return not found

Reply:
Yes, I agree. Since the record was not found, there were no issues with the actual request.